addons/account/controllers/download_docs.py

```diff
@@ -50,8 +51,9 @@ def download_invoice_documents_filetype(self, invoices, filetype, allow_fallback
         invoices.line_ids.check_access('read')
         docs_data = []
         for invoice in invoices:
-            doc_data = invoice._get_invoice_legal_documents(filetype, allow_fallback=allow_fallback)
-            if doc_data:
+            if doc_data := invoice._get_invoice_legal_documents(filetype, allow_fallback=allow_fallback):
+                if (errors := doc_data.pop('errors', False)) and len(invoices) == 1:
```

Comment:
```suggestion
                if (errors := doc_data.get('errors')) and len(invoices) == 1:
```

Reply:
The pop() was intended to avoid to modify 
`headers = _get_headers(**doc_data)` -> `def _get_headers(filename, filetype, content):` 
(there will be one extra key that fails to enter the signature)
I can modify the `_get_header` if you prefer :slightly_smiling_face: